packages/did-provider-oyd/src/oyd-did-provider.ts

```diff
@@ -58,46 +62,141 @@ export class OydDIDProvider extends AbstractIdentifierProvider {
         },
       },
       context
-    )
+    );
 
     const identifier: Omit<IIdentifier, 'provider'> = {
       did: didDoc.did,
       controllerKeyId: key.kid,
       keys: [key],
       services: [],
+    };
+    debug('Created', identifier.did);
+    return identifier;
+  }
+
+  async createIdentifierWithCMSM(
+    { kms, options }: { kms?: string, options: OydCreateIdentifierOptions },
+    context: IContext
+  ): Promise<Omit<IIdentifier, 'provider'>> {
+    if(!this.cmsmOptions) throw new Error("did:oyd: no cmsm options defined!!");
+
+    const createIdentifier = 'https://oydid-registrar.data-container.net/1.0/createIdentifier';
+
+    const pubKey = this.cmsmOptions?.publicKeyCallback("default", "local"); // "default" is probably not right, TODO!!
+    const kid = pubKey.kid;
+    const keyType = pubKey.type;
+
+    const body_create = {  // specify the Identifier options for the registar
+      "key": kid,
+      "options": {
+        "cmsm": true,
+        "key_type": keyType
+      }
+    };
+
+    let signValue: any | undefined;  // do the request
+    try {
+      const response = await fetch(createIdentifier, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body_create),
+      });
+      if (!response.ok) {
+        throw new Error('Network response was not ok: ' + response.statusText);
+      }
+      signValue = await response.json();
+    } catch (error) {
+      // @ts-ignore
+      throw new Error('There has been a problem with the fetch operation: ' + error.toString());
     }
-    debug('Created', identifier.did)
-    return identifier
+    
+    // we received our value to sign, now we sign it!
+    const { sign } = signValue;
+    const signature = this.cmsmOptions.signCallback(kid, sign);
+
+    const body_signed = {
+      "key": kid,
+      "options": {
+        "cmsm": true,
+        "sig": signature
+      }
+    };
+
+    Object.assign(body_signed.options, options);
+
+    let didDoc: any | undefined;  // do the request
+    try {
+      const response = await fetch(createIdentifier, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body_signed),
+      });
+      if (!response.ok) {
+        throw new Error('Network response was not ok: ' + response.statusText);
+      }
+      didDoc = await response.json();
+    } catch (error) {
+      // @ts-ignore
+      throw new Error('There has been a problem with the fetch operation: ' + error.toString());
+    }
+
+    let oydKeyType: OydDidSupportedKeyTypes = "Ed25519";  // make this not static, TODO!!
```

Comment:
This seems problematic to me. Especially since we do not support ed25519 in hardware. Can this be added to the options?. Furthermore shouldn't this simply take the supplied keyType into account?